edi_oca/models/edi_exchange_record.py

```diff
@@ -187,6 +202,66 @@ def _compute_ack_expected(self):
         for rec in self:
             rec.ack_expected = bool(self.type_id.ack_type_id)
 
+    def _get_edi_first_related_record_query(self):
+        query = """
+            WITH related_records AS (
+                SELECT
+                    exchange_record_id,
+                    res_id,
+                    model,
+                    ROW_NUMBER() OVER (PARTITION BY exchange_record_id ORDER BY id) AS row_num
+                FROM
+                    edi_exchange_related_record
+                WHERE
+                    exchange_record_id in %s
+            )
+            SELECT
+                exchange_record_id,
+                res_id,
+                model
+            FROM
+                related_records
+            WHERE
+                row_num = 1;
+        """
+        params = (tuple(self.ids),)
+        return query, params
+
+    def _get_edi_first_related_record_dict(self):
+        """
+        Returns a dictionary with exchange_record_id as key and
+        the first related record res_id and model as value.
+        """
+        self.env["edi.exchange.related.record"].flush_model()
+        query, params = self._get_edi_first_related_record_query()
+        self.env.cr.execute(query, params)
+        return {
+            rec[0]: {"res_id": rec[1], "model": rec[2]}
+            for rec in self.env.cr.fetchall()
+        }
+
+    @api.depends("related_record_ids")
+    def _compute_record(self):
+        """
+        Main related record is computed as the first related record created
+        for the exchange.
+        """
+        data = self._get_edi_first_related_record_dict()
+        for rec in self:
+            rec.res_id = data.get(rec.id, {}).get("res_id", False)
+            rec.model = data.get(rec.id, {}).get("model", False)
+
+    def _inverse_record(self):
+        """
+        Create related record when writing the main record for the exchange.
+        """
+        for rec in self:
+            if not rec.res_id or not rec.model or rec.related_record_ids:
```

Comment:
so you assume that if you have `related_record_ids` valued you don't need to check if this record is already there?

Reply:
you are right, better to check if it is there in case it needs to be added 👍🏼